tools/cordon-cluster/main/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils/condition"
+	"go.goms.io/fleet/tools/utils"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	ctx    = context.Background()
+)
+
+const (
+	testEvictionNameFormat = "test-eviction-%s-%s"
+)
+
+func main() {
+	hubClusterContext := flag.String("hubClusterContext", "", "the kubectl context for the hub cluster")
+	clusterName := flag.String("clusterName", "", "name of the cluster to cordon")
+	flag.Parse()
+
+	if *hubClusterContext == "" {
+		log.Fatalf("hub cluster context for kubectl cannot be empty")
+	}
+	if *clusterName == "" {
+		log.Fatalf("cluster name to cordon cannot be empty")
+	}
+
+	if err := clusterv1beta1.AddToScheme(scheme); err != nil {
+		log.Fatalf("failed to add custom APIs (cluster) to the runtime scheme: %v", err)
+	}
+	if err := placementv1beta1.AddToScheme(scheme); err != nil {
+		log.Fatalf("failed to add custom APIs (placement) to the runtime scheme: %v", err)
+	}
+
+	hubClient, err := utils.GetClusterClientFromClusterContext(*hubClusterContext, scheme)
+	if err != nil {
+		log.Fatalf("failed to create hub cluster client: %v", err)
+	}
+
+	var mc clusterv1beta1.MemberCluster
+	if err = hubClient.Get(ctx, types.NamespacedName{Name: *clusterName}, &mc); err != nil {
+		log.Fatalf("failed to get member cluster %s: %v", *clusterName, err)
+	}
+
+	var crpList placementv1beta1.ClusterResourcePlacementList
+	if err = hubClient.List(ctx, &crpList); err != nil {
+		log.Fatalf("failed to list cluster resource placements: %v", err)
+	}
+
+	// find all unique CRP names for which eviction needs to occur.
+	crpNameMap := make(map[string]bool)
+	for i := range crpList.Items {
+		for j := range crpList.Items[i].Status.PlacementStatuses {
+			if crpList.Items[i].Status.PlacementStatuses[j].ClusterName == *clusterName {
```

Comment:
This doesn't mean the binding exists, what happens if it doesn't exist. What about binding is not bound?